crates/rethnet_evm_napi/src/block.rs

```diff
@@ -261,7 +262,10 @@ impl TryFrom<BlockHeader> for rethnet_eth::block::Header {
             base_fee_per_gas: value
                 .base_fee_per_gas
                 .map_or(Ok(None), |fee| fee.try_cast().map(Some))?,
-            withdrawals_root: value.withdrawals_root.map(|root| B256::from_slice(&root)),
+            withdrawals_root: value
+                .withdrawals_root
+                .map(TryCast::<B256>::try_cast)
+                .transpose()?,
```

Comment:
What does the `transpose` do here?

Reply:
It turns the `Option<Result<T, E>>` into a `Result<Option<T>, E>>`